lib/Parser/Parser.yy

```diff
@@ -632,23 +810,53 @@ unique_shared_opt:
 
 literal:
       boolean_literal
-    | intLit
-    | floatLit
-    | stringLit
+    | intLit { 
+          int intVal = std::stoi($1.getLexeme().str());
+          $$ = ctx.getASTMemoryArena().create<LiteralExpr>(
+                      llvm::APInt(32, intVal),
+                      ctx.getTypesMemoryArena().create<IntTy>(IntTy::Signed, 32),
+                      sm.getSourceLocFromToken($1));
+          std::cerr << "Parsed int literal: " << intVal << std::endl;
+      }
+    | floatLit { 
+          float floatVal = std::stof($1.getLexeme().str());
+          $$ = ctx.getASTMemoryArena().create<LiteralExpr>(
+                      llvm::APFloat(floatVal),
+                      ctx.getTypesMemoryArena().create<FloatTy>(FloatTy::FLOAT),
+                      sm.getSourceLocFromToken($1));
+          std::cerr << "Parsed float literal: " << floatVal << std::endl;
+      }
+    | stringLit { 
+          float floatVal = std::stof($1.getLexeme().str());
+          $$ = ctx.getASTMemoryArena().create<LiteralExpr>(
+                      llvm::APFloat(floatVal),
+                      ctx.getTypesMemoryArena().create<FloatTy>(FloatTy::FLOAT),
+                      sm.getSourceLocFromToken($1));
+          std::cerr << "Parsed string literal: " << floatVal << std::endl;
+      }
     ;
 
 boolean_literal:
-      trueKw
-    | falseKw
+    | trueKw { $$ = ctx.getASTMemoryArena().create<LiteralExpr>(true, ctx.getTypesMemoryArena().create<BoolTy>(), sm.getSourceLocFromToken($1)); }
+    | falseKw { $$ = ctx.getASTMemoryArena().create<LiteralExpr>(false, ctx.getTypesMemoryArena().create<BoolTy>(), sm.getSourceLocFromToken($1)); }
     ;
 
 namespaced_identifier:
       ident ns_id_list_tail
+      {
+        std::string fullName = $1.getLexeme().str() + $2;
+        $$ = ctx.getASTMemoryArena().create<RefExpr>(
+                sm.getSourceLocFromToken($1), fullName);
```

Comment:
RefExpr takes a StringRef, you're creating a pointer to a temporary. Will crash.